### ISSUE TITLE
invalid conversion from ‘const AVCodec*’ to ‘AVCodec*’ with system ffmpeg on multiple distros (FFmpeg 5.1.4 and 6.0))

### ISSUE THREAD

**Issue by theofficialgman** [hrydgard/ppsspp#18672]
### Platform

Linux / BSD

### Compiler and build tool versions

GCC 12.2 + CMAKE 3.25 (Bookworm)

### Operating system version

Debian Bookworm

### Build commands used

```bash
#!/bin/bash

version=v1.16.6

# remove deprecated files
rm -f ~/.local/share/applications/ppsspp.desktop ~/ppsspp

cd /tmp || error "Could not move to /tmp folder"
install_packages build-essential cmake libgl1-mesa-dev libsdl2-dev libsdl2-ttf-dev libvulkan-dev libfontconfig1-dev qt5-qmake \
  qtmultimedia5-dev qttools5-dev-tools libqt5opengl5-dev libcurl4-openssl-dev qtmultimedia5-dev libavcodec-dev \
  libavdevice-dev libavfilter-dev libavformat-dev libavutil-dev libpostproc-dev libswresample-dev libswscale-dev || exit 1
rm -rf ppsspp
git clone https://github.com/hrydgard/ppsspp.git --recurse-submodules --shallow-submodules --depth=1 -b "$version" || error "Failed to clone the repo"
cd ppsspp || error "Could not move to ppsspp folder"
mkdir build || error "Could not make build folder"
cd build || error "Could not move to build folder"
# vulkan does not currently work properly (or at all) in SDL or QT backends on Linux https://github.com/hrydgard/ppsspp/pull/13054 https://github.com/hrydgard/ppsspp/issues/11628
# changing to vulkan in the UI does nothing and OpenGL continues to be used
# enabling it causes no compilation issues and it can not be fully disabled (always shows in the UI) so keep it enabled
cmake .. -DUSING_QT_UI=ON -DUSE_SYSTEM_FFMPEG=ON || error "cmake failed to configure the build"
make -j$(nproc) || error "Could not build ppsspp"
sudo make install || error "Could not install ppsspp"
# update timestamp of top level icon directory to signal icon cache to be refreshed
sudo touch /usr/local/share/icons/hicolor
rm -rf /tmp/ppsspp
```
https://github.com/Botspot/pi-apps/pull/2518

### What happens

Bookworm ARM32: https://github.com/Botspot/pi-apps/actions/runs/7436407610/job/20232824146#step:3:2963
Bookworm ARM64: https://github.com/Botspot/pi-apps/actions/runs/7436407610/job/20232824208#step:3:3035
Mantic ARM64: https://github.com/Botspot/pi-apps/actions/runs/7436407610/job/20232824418#step:3:3603

### PPSSPP version affected

v1.16.6

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] Make sure to run `git submodule update --init --recursive` before building.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [X] Try deleting the build directory and running the build again.
- [ ] Check GitHub Actions, which builds every merge and PR.
- [X] Include logs and help us reproduce.

**Comment by hrydgard** [comment]
Do try to use the bundled ffmpeg instead of system ffmpeg. We very intentionally ship a fixed version of ffmpeg that supports everything we'll ever need and doesn't change.

**Comment by hrydgard** [comment]
https://github.com/hrydgard/ppsspp/pull/18670 will probably fix this though.

**Comment by theofficialgman** [comment]
> Do try to use the bundled ffmpeg instead of system ffmpeg. We very intentionally ship a fixed version of ffmpeg that supports everything we'll ever need and doesn't change.

bundled system ffmpeg is not patched for hardware video decoding on the intended platforms so it is not desirable to use as it will use software video decoding.

**Comment by theofficialgman** [comment]
indeed it appears that the hardware video decoding of the videos I have tested just immediately get skipped likely to no supporting that non-standard resolution. I will switch to internal ffmpeg and see if it can be built properly still on all intended systems.

**Comment by Tatsh** [comment]
> > Do try to use the bundled ffmpeg instead of system ffmpeg. We very intentionally ship a fixed version of ffmpeg that supports everything we'll ever need and doesn't change.
> 
> bundled system ffmpeg is not patched for hardware video decoding on the intended platforms so it is not desirable to use as it will use software video decoding.

ffmpeg by default uses software decoding. Are you using a system version that is patched to always attempt hardware?

**Comment by theofficialgman** [comment]
> > > Do try to use the bundled ffmpeg instead of system ffmpeg. We very intentionally ship a fixed version of ffmpeg that supports everything we'll ever need and doesn't change.
> > 
> > 
> > bundled system ffmpeg is not patched for hardware video decoding on the intended platforms so it is not desirable to use as it will use software video decoding.
> 
> ffmpeg by default uses software decoding. Are you using a system version that is patched to always attempt hardware?

I'm sure because I patched it to be so (on Nintendo Switch) and other distro maintainers do so as well for their hardware (eg: rockchip, broadcom, etc)

**Comment by Tatsh** [comment]
This should be resolved now that #18670 is merged.
